icis-air-inspections.ipynb

```diff
@@ -545,66 +373,32 @@
-      "text/plain": [
-       "<Figure size 1080x720 with 1 Axes>"
-      ]
-     },
-     "metadata": {
-      "needs_background": "light"
-     },
-     "output_type": "display_data"
-    }
-   ],
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
    "source": [
     "# This cell creates a function that will be used by both the CD and the state\n",
     "# to plot the number of cases by year.\n",
     "import datetime\n",
     "\n",
-    "def show_plot( df, date_field, year_field, place, date_format, chart_title ):\n",
-    "    format_str = date_format # The format\n",
-    "    nan_count = 0\n",
-    "    year_col = []\n",
-    "    for day in df[date_field]:\n",
-    "        try:\n",
-    "            viol_year = datetime.datetime.strptime(day, format_str).year\n",
-    "            year_col.append( viol_year )\n",
-    "        except:\n",
-    "            nan_count += 1\n",
-    "            year_col.append(  np.NaN )\n",
-    "    df[year_field] = year_col\n",
+    "def show_plot( df, date_field, place, date_format, chart_title ):\n",
     "    \n",
-    "    year_groups = df.groupby( year_field )[[ year_field ]]\n",
-    "    counted_years = year_groups.count()\n",
+    "    df[date_field] = pd.to_datetime(df[date_field], format=date_format) #format\n",
     "\n",
-    "    # Print how many values are present \n",
-    "    print(counted_years)\n",
+    "    cy = df.groupby([date_field])['ACTUAL_END_DATE'].count() # Use a dummy variable to count inspections.\n",
+    "    cy = cy.resample('Y').sum() # resample specific days to a yearly basis  \n",
+    "    cy = pd.DataFrame(cy) # convert to dataframe\n",
+    "    cy.index = cy.index.strftime('%Y') # makes the x axis (date) prettier\n",
+    "    \n",
     "    chart_title +=  \" in \" + place + \" by year\"\n",
     "\n",
-    "    ax = counted_years[[year_field]].plot(kind='bar', title = chart_title, figsize=(15, 10), legend=False, fontsize=12)\n",
+    "    ax = cy.plot(kind='bar', title = chart_title, figsize=(15, 10), legend=False, fontsize=12)\n",
```

Comment:
when there is only one inspection as the max value, left axis shows as floats, which is pretty weird. Can we tell it to use whole-number steps only?

Reply:
Yes, but the easiest way I've seen to do this involves importing another module. So, will do, but maybe in a separate issue.